File: longestCommonPrefix.java
class longestCommonPrefix {
    public static String longestCommonPrefix(String[] strs){

        if (strs.length==0){
            return "";
        } else if(strs.length==1){
            return strs[0];
        }
        int shortest=0;
        int i;

        for (i=1;i<strs.length;i++){
            if (strs[i].length() < strs[shortest].length()){
                shortest=i;
            }
        }

        if (strs[shortest].length()==0){
            return "";
        }

        int letter=0;
        char[] prefix=new char[strs[shortest].length()];
        boolean broke=false;

        while(letter<strs[shortest].length()){

            char cur=strs[0].charAt(letter);

            for (i=1;i<strs.length;i++){
                if (strs[i].charAt(letter) != cur){
                    broke=true;
                    break;
                }
            }

            if (broke) {
                break;
            }

            prefix[letter]=cur;

            letter++;
        }

        if (prefix[0]=='\0'){
            return "";
        }

        char[] prefNew = new char[letter];

        for (i=0;i<letter;i++){
            prefNew[i]=prefix[i];
        }

        return String.valueOf(prefNew);
    }

    public static void main(String[] args) {
        String[] strs={"",""};
        String str = longestCommonPrefix(strs);
        System.out.println(str);
    }
}

File: longest_common_prefix.py
class Longest_common_prefix:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        '''
        Trie: children, char
        root: [Tries](children)
        '''

        root=Trie('0')
        cur=root
        maxLen=2**32-1
        for i,st in enumerate(strs):
            leng=0
            for ind,car in enumerate(st):
                if ind>=maxLen:
                    break
                if not car in cur.childTrie:
                    if i!=0:
                        break
                    cur.childTrie[car]=Trie(car)
                cur=cur.childTrie[car]
                leng+=1
            cur=root
            maxLen=min(leng,maxLen)
        return strs[0][0:maxLen]

class Trie:
    car='0'
    childTrie={}

    def __init__(self,car):
        self.car=car
        self.childTrie={}
        